### ISSUE TITLE
geometry failure on windows, works on web.

### ISSUE THREAD

**Issue by Justinfront** [underscorediscovery/snow#75]
I have updated snow and luxe this afternoon and updated one of the hxDaedalus Luxe examples ( changing image loading around ) but it fails at runtime on windows but succeeds on web I can dig deeper into how to create ndll etc... but if you know the fix that would be amazing. Details of error and how to reproduce below.

_________________________________________________________________________
**...\bin\windows.build\cpp\src\luxe\structural\BalancedBSTNode_phoenix_geometry_GeometryKey_phoenix_geometry_Geometry.cpp : fatal error C1083: Cannot open compiler generated file: 'obj/msvc18-ncxp/04e3d618_BalancedBSTNode_phoenix_geometry_GeometryKey_phoenix_geometry_Geometry.obj': No such file or directory**

**BalancedBST_phoenix_geometry_GeometryKey_phoenix_geometry_Geometry.cpp
BalancedBSTTraverseMethod.cpp**
**GeometryUtils.cpp**
**flow /*8
**build - stopping because of errors in hxcpp compile, while building arch 32**

**Error: Command failed with error 1**
_________________________________________________________________________
To run the code you will need latest hxDaedalus:
haxelib git hxDaedalus https://github.com/hxDaedalus/hxDaedalus
and then modify this test: 
https://github.com/hxDaedalus/hxDaedalus-luxe-examples/tree/master/hxDaedalus-Luxe-Examples/07-MeshExtractionFromBitmap

with the following code.

_________________________________________________________________________
```haxe
///
package;

import hxDaedalus.data.math.Tools;
import hxDaedalus.graphics.TargetCanvas;
import hxDaedalus.graphics.SimpleDrawingContext;
import hxDaedalus.data.math.Point2D;
import luxe.Input.Key;
import luxe.Input.KeyEvent;
import luxe.Sprite;
import luxe.Vector;
import phoenix.Texture;


class Main extends luxe.Game {
	
	// entry point
    override function ready() {

		// white background
		Luxe.renderer.clear_color.rgb(0xFFFFFF);
		
        // get the asset
		var texture = Luxe.resources.texture('assets/MeshExtractionFromBitmap.png');
		
		// show the image
		var sprite = new Sprite({
			texture: texture,
			centered: false,
			pos: new Vector(110, 220),
			size: new Vector( texture.width, texture.height ),
			depth: -1
		});
		
        var vertices = new Array<Point2D>();
        var triangles = new Array<Int>();
		Tools.extractMeshFromBitmap( texture, vertices, triangles);
		
        // create a viewport
		var g = new SimpleDrawingContext( new TargetCanvas() );
		var dx = 310;
		var dy = 220;
		g.lineStyle(1, 0xFF0000);
		// use a 3 iterator instead?
		var i: Int = 0;
		while( i < triangles.length ){
			g.moveTo(vertices[triangles[i]].x + dx, vertices[triangles[i]].y + dy);
		   	g.lineTo(vertices[triangles[i+1]].x + dx, vertices[triangles[i+1]].y + dy);
		    g.lineTo(vertices[triangles[i+2]].x + dx, vertices[triangles[i+2]].y + dy);
		    g.lineTo(vertices[triangles[i]].x + dx, vertices[triangles[i]].y + dy);
			i+=3;
	    }
		
		
	}
	
	override function onkeyup(e:KeyEvent) {

        if (e.keycode == Key.escape) {
            Luxe.shutdown();
        }

    } // onkeyup

		// standard setup type stuff
	override function config( config: luxe.AppConfig ) {
		
		config.preload.textures = [
            { id:'assets/MeshExtractionFromBitmap.png' }
        ];
		config.render.antialiasing = 4;
		return config;
	}
	
} // Main
```_______________________________________________________________________

**Comment by underscorediscovery** [comment]
The error looks like a MSVC compiler error, a clean build may be needed.
Try deleting the bin folder (or build with `flow run --clean` which does the same thing).

**Comment by Justinfront** [comment]
I definitely deleted the bin folder at least once today.
But tomorrow morning I can try the flow run -clean.
The code works fine for luxe web target and I am pretty sure we are not doing anything specifically in our code different for luxe web and windows targets for the geometry code, and the code worked fine on the 5th of April and the only modifications made are related to luxe pixels I don't believe any have been made in relation to geometry.
So at the moment I am leaning towards a luxe bug, but I can try to dig a bit deeper into luxe code I am unsure how to create the ndll do you have some clues on how that all works and how the c externs are created etc...

**Comment by azrafe7** [comment]
@Justinfront, actually could be the same problem I have on my windows machine (discussed it a few days ago on gitter with @underscorediscovery). To see if it's it please try running the Luxe rendertexture demo (in `luxe\git\tests\rendering\rendertexture`), which for me doesn't work when targeting native, but does with web.

**Comment by Justinfront** [comment]
Just tried it on mac, actually I went through all these already on the mac, trying to rem when I last updated git I think it was very recent.  I can try it on my thinkpad but not even sure I have luxe on there yet, or wait till tomorrow at work.

**Comment by underscorediscovery** [comment]
The errors can't be related to mac, this is explicitly an MSVC error:
` fatal error C1083: Cannot open compiler generated file` MSVC itself is creating a file, and then unable to read it during the same compile.

**Comment by Justinfront** [comment]
haxelib run flow run --clean
gives the same error on the work machine.

( I did not yet finish setting up the thinkpad at home I might take windows off it as it was behaving rather badly maybe it was just annoyed I neglected it, but more likely it's the lenovo bloatware but tricky to know what parts I can remove, but looks like I need to spend some serious time cleaning it up :( installed latest haxe and git's for snowkit and latest hxcpp, but I think after spybot had finished and some of the haxelib git's were slow, I did not yet put snow.ndll as I was not sure if that happened magically yet,but I can give that a try tonight but it might be late as I have tango lesson and there is a live salsa band in pub ).

In the meantime I guess I should look into building snow.ndll myself.

**Comment by Justinfront** [comment]
Ah found the relevant rebuild page..http://underscorediscovery.github.io/snow/guide/native-layer.html

**Comment by Justinfront** [comment]
Ok rebuild is not helping, it seems to be called libsnow.lib rather than snow.ndll ?

**Comment by Justinfront** [comment]
That's interesting if I use --arch 64 on both I get:

**flow / build - running haxe ...**
**Sys_error("bin/windows64.build/cpp/include/luxe/structural/BalancedBSTNode_phoenix_geometry_GeometryKey_phoenix_geometry_Geometry.h: No such file or directory")**

Which seems odd, because the file exists for the 32bit.

**Comment by Justinfront** [comment]
Looking in the obj...
04e3d618_BalancedBST_phoenix_geometry_GeometryKey_phoenix_geometry_Geometry.obj
or 
04e3d618_BalancedBSTNode.obj
is not the same as 
04e3d618_BalancedBSTNode_phoenix_geometry_GeometryKey_phoenix_geometry_Geometry.obj

**Comment by Justinfront** [comment]
cl.exe -Iinclude -nologo -O2 /WX- /fp:precise -DHX_WINDOWS -D_USING_V120_SDK71_ -GR -Oy- -c -EHs -GS- -arch:SSE -IC:/HaxeToolkit/haxe/lib/hxcpp/3,2,94
/include -DHXCPP_VISIT_ALLOCS -DHXCPP_API_LEVEL=321 -D_CRT_SECURE_NO_DEPRECATE -D_ALLOW_MSC_VER_MISMATCH -D_ALLOW_ITERATOR_DEBUG_LEVEL_MISMATCH -DHX_W
IN_MAIN -wd4996 -MT ./src/luxe/structural/BalancedBSTNode_phoenix_geometry_GeometryKey_phoenix_geometry_Geometry.cpp -Foobj/msvc18-ncxp/04e3d618_Balan
cedBSTNode_phoenix_geometry_GeometryKey_phoenix_geometry_Geometry.obj
OrderedMapIterator.cpp
cl.exe -Iinclude -nologo -O2 /WX- /fp:precise -DHX_WINDOWS -D_USING_V120_SDK71_ -GR -Oy- -c -EHs -GS- -arch:SSE -IC:/HaxeToolkit/haxe/lib/hxcpp/3,2,94
/include -DHXCPP_VISIT_ALLOCS -DHXCPP_API_LEVEL=321 -D_CRT_SECURE_NO_DEPRECATE -D_ALLOW_MSC_VER_MISMATCH -D_ALLOW_ITERATOR_DEBUG_LEVEL_MISMATCH -DHX_W
IN_MAIN -wd4996 -MT ./src/luxe/structural/BalancedBSTTraverseMethod.cpp -Foobj/msvc18-ncxp/04e3d618_BalancedBSTTraverseMethod.obj
BalancedBST_phoenix_geometry_GeometryKey_phoenix_geometry_Geometry.cpp
cl.exe -Iinclude -nologo -O2 /WX- /fp:precise -DHX_WINDOWS -D_USING_V120_SDK71_ -GR -Oy- -c -EHs -GS- -arch:SSE -IC:/HaxeToolkit/haxe/lib/hxcpp/3,2,94
/include -DHXCPP_VISIT_ALLOCS -DHXCPP_API_LEVEL=321 -D_CRT_SECURE_NO_DEPRECATE -D_ALLOW_MSC_VER_MISMATCH -D_ALLOW_ITERATOR_DEBUG_LEVEL_MISMATCH -DHX_W
IN_MAIN -wd4996 -MT ./src/luxe/structural/BalancedBSTNode.cpp -Foobj/msvc18-ncxp/04e3d618_BalancedBSTNode.obj
BalancedBSTNode_phoenix_geometry_GeometryKey_phoenix_geometry_Geometry.cpp
C:\projects\currentProjects\experimental\other\old\addCurveTo\hxDaedalus-luxe-examples\hxDaedalus-Luxe-Examples\07-MeshExtractionFromBitmap\bin\window
s.build\cpp\src\luxe\structural\BalancedBSTNode_phoenix_geometry_GeometryKey_phoenix_geometry_Geometry.cpp : fatal error C1083: Cannot open compiler g
enerated file: 'obj/msvc18-ncxp/04e3d618_BalancedBSTNode_phoenix_geometry_GeometryKey_phoenix_geometry_Geometry.obj': No such file or directory

**Comment by Justinfront** [comment]
I think the error might relate to 
luxe.debug.BatcherDebugView 

import luxe.structural.BalancedBST.BalancedBSTNode;

is this the correct path for a class in BalancedBST.  Is DCE removing the BalanecdBSTNode ? Can I disable BST , I tried moving the Class separate but still the object is not created?

**Comment by Justinfront** [comment]
Ok this is really nasty issue and seems to be a windows7 bug!!!  The problem was that obj files are not created if the path, and file name are too long!!!  
I moved project to a shorter folder structure and it worked fine.  Sorry for this issue - not something I would have expected, I only found it by trying to manually copy the obj file from an older project, but maybe some thought to strategies to keep obj file names from getting too long might help?
I should raise the bug on hxcpp so that maybe hxcpp can warn user?
